naslib/defaults/trainer.py

```diff
@@ -108,11 +113,26 @@ def search(self, resume_from="", summary_writer=None, after_epoch: Callable[[int
 
         for e in range(start_epoch, self.epochs):
 
+            # create the arch directory (without overwriting)
+            if self.config.save_arch_weights: 
+                Path(f"{self.config.save_arch_weights_path}/epoch_{e}").mkdir(parents=True, exist_ok=False)
+
             start_time = time.time()
             self.optimizer.new_epoch(e)
 
+            arch_weights_lst = []
```

Comment:
Generally, as a programming practice, don't mention the data-structure in the name of the variable. `arch_weights.append(...)` makes it clear that it is a list, not a dict.